### PR TITLE
Faster registry

### PR DESCRIPTION
**Pull request by bsless**
Speed up registry creation and usage

Part of #513

### CODE REVIEW

**Comment by bsless** [comment]
```clojure
(def regs
  {:nil nil
   :map {}
   :simple (mr/simple-registry {})
   :composite (mr/composite-registry (mr/simple-registry {}))
   :mutable (mr/mutable-registry (atom {}))})

(doseq [[name r] regs]
  (let [mean (first (:mean (cc/quick-benchmark (mr/registry r) {})))
        [scale unit] (cc/scale-time mean)]
    (println name (* mean scale) unit)))

;;; Before
;; :nil 11.314252099384198 ns
;; :map 11.553363773721745 µs
;; :simple 5.894620337443649 ns
;; :composite 5.851375239599097 ns
;; :mutable 5.893465716928456 ns

;;; After
;; :nil 3.259309437187958 ns
;; :map 22.65670884744675 ns
;; :simple 3.267212075708806 ns
;; :composite 3.287585829320794 ns
;; :mutable 3.289205443217265 ns
```

**Comment by bsless** [comment]
Also think I found a mistake in one of the tests

**Comment by ikitommi** [comment]
thanks!

File: src/malli/registry.cljc [hunk 1]
```diff
@@ -11,12 +11,22 @@
...
   (reify
     Registry
-    (-schema [_ type] (get schemas type))
+    (-schema [_ type] (schemas type))
     (-schemas [_] schemas)))
 
-(defn registry [?registry]
-  (cond (satisfies? Registry ?registry) ?registry
-        (map? ?registry) (simple-registry ?registry)))
+(defn registry
```
Comment (ikitommi) [inline src/malli/registry.cljc:17]: simplified this into:

```clj
(defn registry [?registry]
  (cond (nil? ?registry) nil
        #?@(:clj [(instance? malli.registry.Registry ?registry) ?registry])
        (map? ?registry) (simple-registry ?registry)
        (satisfies? Registry ?registry) ?registry))
```

`instance?` -check works with reified classes? woot.

Comment (bsless) [reply src/malli/registry.cljc:17]: They do. I dug into why `satisfies?` is fast for reified classes, and voila!
I also found that `nil?` checks are slightly slower than raw `if`

**Pull request merged**